crates/datasources/src/lance/mod.rs

```diff
@@ -76,114 +69,69 @@ impl TableProvider for LanceTable {
         input: Arc<dyn ExecutionPlan>,
         _overwrite: bool,
     ) -> Result<Arc<dyn ExecutionPlan>> {
-        Ok(Arc::new(LanceInsertExecPlan::new(
+        Ok(Arc::new(self::insert::LanceInsertExecPlan::new(
             self.dataset.clone(),
             input,
         )))
     }
 }
 
-struct LanceInsertExecPlan {
-    dataset: Dataset,
-    input: Arc<dyn ExecutionPlan>,
-    metrics: ExecutionPlanMetricsSet,
-}
-
-impl LanceInsertExecPlan {
-    fn new(dataset: Dataset, input: Arc<dyn ExecutionPlan>) -> Self {
-        Self {
-            dataset,
-            input,
-            metrics: ExecutionPlanMetricsSet::new(),
-        }
-    }
-}
-
-impl DisplayAs for LanceInsertExecPlan {
-    fn fmt_as(&self, _t: DisplayFormatType, f: &mut std::fmt::Formatter) -> std::fmt::Result {
-        write!(f, "LanceInsertExecPlan")
-    }
-}
-
-impl std::fmt::Debug for LanceInsertExecPlan {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(f, "LanceInsertExecPlan: {:?}", self.schema())
-    }
-}
-
-impl ExecutionPlan for LanceInsertExecPlan {
-    fn as_any(&self) -> &dyn Any {
-        self
-    }
-
-    fn schema(&self) -> SchemaRef {
-        COUNT_SCHEMA.clone()
-    }
 
-    fn output_partitioning(&self) -> Partitioning {
-        Partitioning::UnknownPartitioning(1)
-    }
+pub struct LanceDatasource;
 
-    fn output_ordering(&self) -> Option<&[PhysicalSortExpr]> {
-        None
+#[async_trait]
+impl Datasource for LanceDatasource {
+    fn name(&self) -> &'static str {
+        "lance"
     }
 
-    fn children(&self) -> Vec<Arc<dyn ExecutionPlan>> {
-        Vec::new()
-    }
+    /// Create a new datasource from the provided table options and credentials.
+    /// CREATE EXTERNAL TABLE foo FROM <name> OPTIONS (...) [CREDENTIALS] (...) [TUNNEL] (...)
+    // TODO: the datasource should have control over it's own CredentialsOptions and TunnelOptions
```

Comment:
What does this mean?

Reply:
Oh i forgot to remove this. This made more sense in a previous iteration.